### PR TITLE
Port a few more endpoints to the typed API

### PR DESCRIPTION
**Pull request by danvk**
I filled in the FastAPI / Pydantic schemas in a few places. Note that POST requests need a JSON payload with this system. If we want to support empty payloads, we'll need to add some more logic in `typed-api.ts`.

### CODE REVIEW

**Comment by cguedes** [comment]
> Note that POST requests need a JSON payload with this system. 

I agree that POST should alway have a JSON payload, even though sometimes it can be empty.

**Review by cguedes — APPROVED** [review]
Approved with a note about the create project payload (querystring vs post body). We should update that later.

**Review by sehyod — APPROVED** [review]

File: python/sidecar/http.py [hunk 1]
```diff
@@ -166,7 +168,9 @@ async def list_projects():
 
 
 @project_api.post("/")
-async def create_project(project_name: str, project_path: str = None):
+async def create_project(
+    payload: EmptyRequest, project_name: str, project_path: str = None
```
[the remark this answers is not in the document]
Comment (cguedes) [reply python/sidecar/http.py:172]: I did this change when I merged the from main.

**Review by cguedes — APPROVED** [review]

File: python/sidecar/http.py [hunk 2]
```diff
@@ -334,3 +328,4 @@ async def get_settings() -> FlatSettingsSchema:
 async def update_settings(req: FlatSettingsSchemaPatch) -> FlatSettingsSchema:
     user_id = "user1"
     return settings.update_settings_for_user(user_id, req)
+    return settings.update_settings_for_user(user_id, req)
```
Comment (gjreda) [inline python/sidecar/http.py:331]: This shouldn't have been merged

Comment (cguedes) [reply python/sidecar/http.py:331]: Nice catch 😅

Comment (danvk) [reply python/sidecar/http.py:331]: I've noticed this happen on a few of our Python files. I think something that runs when we save Python files must be doing it. Total shot in the dark but maybe it has something to do with missing trailing newlines?

**Pull request merged**